doc/_data/schemas/State/Message.json

```diff
@@ -4,8 +4,12 @@
   "type": "object",
   "properties":
   {
-    "message": { "type": "string", "description": "Message to display" },
-    "type": { "enum": ["info", "success", "warning", "error"], "type": "string", "default": "info" },
+    "message": { "type": "string", "default": "", "description": "Message to display" },
+    "prefix": { "type": "string", "default": "State name", "description": "Prefix message, name of the GUI label" },
+    "log": { "type": "boolean", "default": true, "description": "True to display in the terminal" },
+    "log_type": { "enum": ["info", "success", "warning", "error"], "type": "string", "default": "info", "description": "Type of message to log (corresponding to the LOG_* macros)" },
+    "gui": { "type": "boolean", "default": false, "description": "True to display in the GUI" },
+    "gui_category": { "type": "array", "items": { "type": "string" }, "description": "Category in which the message will be added (when empty, displays at the top of the GUI)" },
```

Comment:
I'm not a huge fan of having to specify two entries to get one behaviour, so I suggest:
- only keep `log`, defaults to an empty string, if empty then no console logging happens, otherwise same effect as `log_type` (note: keep `type` working as before if present in the configuration)
- only keep `gui`, if absent from the configuration then no GUI, otherwise same effect as `gui_category`

Reply:
The idea was to allow showing a message in both the GUI and terminal, or in either one. Sometimes it only makes sense to show things in the GUI (some action to confirm manually for the FSM for instance), sometimes only in the terminal (mostly discrete events), or in both.

If I understand you correctly, what you're suggesting would prevent from displaying a message in the GUI only. I didn't want to have to clutter the terminal with GUI information. Though that's probably not that big a deal, and we could settle for always displaying in the terminal, and only if requested adding it to the GUI. Might even be better, as popping up new categories in the GUI could lead to somewhat hidden messages that one might miss ;)

Reply:
No, what I mean is that instead of:
```yaml
# Log + GUI
log: true
log_type: debug
gui: true
gui_category: [Something]
---
# GUI only
log: false
gui: true
gui_category: [Something]
---
# Log only
log: true
log_type: debug
gui: false
```

We have:
```yaml
# Log + GUI
log: debug
gui: [Something]
---
# GUI only
gui: [Something]
---
# Log only
log: debug
```

Reply:
Ah I see, and that's indeed much better. Thanks for the suggestion.